examples/src/main/scala/org/parboiled2/examples/Calculator2.scala

```diff
@@ -72,24 +72,24 @@ class Calculator2(val input: ParserInput) extends Parser {
   def Expression: Rule1[Expr] =
     rule {
       Term ~ zeroOrMore(
-        '+' ~ Term ~> Addition.apply
-          | '-' ~ Term ~> Subtraction.apply
+        '+' ~ Term ~> Addition.apply _
```

Comment:
Is that unavoidable? I guess it's good for now, but would be nice if we could avoid it in the future.

Reply:
```
$ scala -Xsource:3
Welcome to Scala 2.12.15 (OpenJDK 64-Bit Server VM, Java 1.8.0_302).
Type in expressions for evaluation. Or try :help.

scala> case class Foo(x: Int)
defined class Foo

scala> Foo.apply
<console>:14: error: missing argument list for method apply in object Foo
Unapplied methods are only converted to functions when a function type is expected.
You can make this conversion explicit by writing `apply _` or `apply(_)` instead of `apply`.
       Foo.apply
           ^
```

compile error with Scala 2.12 😢

Reply:
Ok, that's fine then, if we might be able to simplify that later on when 2.12 is dropped eventually, that might be good enough for now. Thanks for checking!